opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -3638,8 +3638,10 @@ namespace Opm
                     total_connection_length += connection_length;
                 }
 
-                SICD& sicd = *segment.spiralICD();
-                sicd.updateScalingFactor(segment_length, total_connection_length);
```

Comment:
We need to do this, probably because (at least, when the code was written), it was not easy to get the information `segment_length` and `total_connection_length` .  If we can do it during parsing, then we do not need to do it now. 

Not sure when PINCH happens, whether it will change the `connection_length`? which is the tricky part.

Reply:
One solution can be we do not store the scaling factor as a member. We just call a function  `scalingFactor()`, but we need to calculate the `total_connection_length` each time.

Reply:
@GitPaean : I am looking into refactoring this. One question: the current code has the lines:
```C++
const int parental_segment_number = segmentSet()[seg].outletSegment();
const double segment_length = segmentSet().segmentLength(parental_segment_number);
```
Why does the `segment_length` variable use the *outlet_segment*?

Reply:
yes. That is the definition. The scaling factor can have a few different ways to calculate, one is based on its parental segment, one is based on its total_completion_length. Please refer to the manual for WSEGSICD for more details. I am not familiar with the physics behind.

Reply:
Ok thank you - I will stare at the manual, to me it looked a bit suspicious.